shared/src/business/useCases/trialSessions/updateTrialSessionInteractor.test.js

```diff
@@ -424,100 +456,225 @@ describe('updateTrialSessionInteractor', () => {
+            closedDate: '2019-03-01T21:42:29.073Z',
+            docketNumber: '888-88',
+            docketNumberWithSuffix: '888-88',
+            hearings: [],
+            status: CASE_STATUS_TYPES.closed,
+            trialDate: '2019-03-01T21:42:29.073Z',
+            trialSessionId: MOCK_TRIAL_ID_4,
+          });
+
+        await updateTrialSessionInteractor(applicationContext, {
+          trialSession: mockInPersonCalendaredTrialSession,
+        });
+
+        expect(
+          applicationContext.getUseCaseHelpers()
+            .setNoticeOfChangeToInPersonProceeding,
+        ).not.toHaveBeenCalled();
+      });
+
+      it('should NOT generate a NOIP when the case status is open, the trial session is calendared but the trial session proceeding type has not changed', async () => {
+        await updateTrialSessionInteractor(applicationContext, {
+          trialSession: mockTrialsById[MOCK_TRIAL_ID_4],
+        });
+
+        expect(
+          applicationContext.getUseCaseHelpers()
+            .setNoticeOfChangeToInPersonProceeding,
+        ).not.toHaveBeenCalled();
+      });
+
+      it('should generate a NOIP when the proceeding type changes from remote to in-person, the case status is not closed, and the trial session is calendared', async () => {
+        const inPersonCalendaredTrialSession = {
+          ...mockTrialsById[MOCK_TRIAL_ID_4],
+          isCalendared: true,
+          proceedingType: TRIAL_SESSION_PROCEEDING_TYPES.inPerson,
+        };
+
+        applicationContext
+          .getPersistenceGateway()
+          .getCaseByDocketNumber.mockReturnValueOnce({
+            ...MOCK_CASE,
+            docketNumber: '888-88',
+            docketNumberWithSuffix: '888-88',
+            hearings: [],
+            trialDate: '2019-03-01T21:42:29.073Z',
+            trialSessionId: MOCK_TRIAL_ID_4,
+          });
+
+        await updateTrialSessionInteractor(applicationContext, {
+          trialSession: inPersonCalendaredTrialSession,
+        });
+
+        expect(
+          applicationContext.getUseCaseHelpers()
+            .setNoticeOfChangeToInPersonProceeding,
+        ).toHaveBeenCalled();
+      });
     });
 
-    expect(
-      applicationContext.getUseCaseHelpers().setNoticeOfChangeOfTrialJudge,
-    ).toHaveBeenCalledTimes(1);
+    describe('Remote Proceeding', () => {
+      it('should NOT generate a NORP when the case status is open, the trial session is calendared but the trial session proceeding type has not changed', async () => {
+        await updateTrialSessionInteractor(applicationContext, {
+          trialSession: mockTrialsById[MOCK_TRIAL_ID_4],
+        });
+
+        expect(
+          applicationContext.getUseCaseHelpers()
+            .setNoticeOfChangeToRemoteProceeding,
+        ).not.toHaveBeenCalled();
+      });
+
+      it('should NOT generate a NORP when the proceeding type changes from in-person to remote, the trial session is calendared but the case is closed', async () => {
```

Comment:
Could add similar tests for 'Change of Trial Judge' now that we're using it in `updateAssociatedCaseAndSetNoticeOfChange`